client/components/settings/add-operation-modal.js

```diff
@@ -55,8 +55,11 @@ export default class AddOperationModal extends React.Component {
 
     shouldComponentUpdate(nextProps, nextState) {
         // Only rerender if the button status has to be updated
-        return this.isSubmitDisabled() ===
-               !(nextState.titleIsOK && nextState.amountIsOK && nextState.dateIsOK);
+        return this.isSubmitDisabled() !== this.willSubmitBeDisabled(nextState);
+    }
+
+    willSubmitBeDisabled(nextState) {
```

Comment:
This code could be shared with `isSubmitDisabled`, if the latter had a `state` parameter.